openedx/core/djangoapps/edx_global_analytics/utils.py

```diff
@@ -0,0 +1,128 @@
+"""
+Helpers for the edX global analytics application.
+"""
+
+import datetime
+import calendar
+
+from django.core.cache import cache
+
+from django.db.models import Count
+from django.db.models import Q
+from student.models import UserProfile
+
+
+def calculate_active_students_amount(query_name, activity_period):
+    """
+    Calculates instance active students amount for particular period as like as previous calendar day,
+    week and month after cached.
+    """
+    period_start, period_end = activity_period
+
+    active_students_amount = UserProfile.objects.exclude(
+        Q(user__last_login=None) | Q(user__is_active=False)
+    ).filter(user__last_login__gt=period_start, user__last_login__lt=period_end).count()
+
+    if query_name is not 'active_students_amount_day':
+        clear_cache_date = period_end
+        return caching_instance_data(query_name, active_students_amount, clear_cache_date)
+
+    return active_students_amount
+
+
+def calculate_students_per_country(query_name, activity_period):
+    """
+    Calculates instance students per country amount for particular period as like as previous calendar day,
+    week and month after cached.
+
+    Returns:
+        students_per_country_after_cached (dict): Country-count accordance as pair of key-value.
+                                                  Example: {u'FR': 3434, u'UA': 1124}
+    """
+    period_start, period_end = activity_period
+
+    students_per_country = dict(UserProfile.objects.exclude(
+        Q(user__last_login=None) | Q(user__is_active=False)
+    ).filter(user__last_login__gt=period_start, user__last_login__lt=period_end
```

Comment:
I researched Django`s logic for that, I`ll implement `__gte` feature, it works fine.